tst/unit/test_container_iterator.cpp

```diff
@@ -67,72 +80,401 @@ TEST_CASE("Can pull variables from containers based on Metadata", "[ContainerIte
-                            int &cnt) { cnt++; },
-              sum);
-          nElements += sum;
-        }
-        REQUIRE(nElements == 40960);
+        using policy4D = Kokkos::MDRangePolicy<Kokkos::Rank<4>>;
+        int total = 0;
+        int sum = 1;
+        Kokkos::parallel_reduce(
+            policy4D({0, 0, 0, 0}, {v.GetDim(4), v.GetDim(3), v.GetDim(2), v.GetDim(1)}),
+            KOKKOS_LAMBDA(const int l, const int k, const int j, const int i, int &cnt) {
+              cnt++;
+            },
+            sum);
+        total += sum;
+        REQUIRE(total == 40960);
       }
     }
 
     WHEN("we set Independent variables to one") {
       // set "Independent" variables to one
-      ContainerIterator<Real> ci(rc, {Metadata::Independent});
-      CellVariableVector<Real> &civ = ci.vars;
-      for (int n = 0; n < civ.size(); n++) {
-        ParArrayND<Real> v = civ[n]->data;
+      auto v = rc.PackVariables({Metadata::Independent});
+      par_for(
+          "Set independent variables", DevExecSpace(), 0, v.GetDim(4) - 1, 0,
+          v.GetDim(3) - 1, 0, v.GetDim(2) - 1, 0, v.GetDim(1) - 1,
+          KOKKOS_LAMBDA(const int l, const int k, const int j, const int i) {
+            v(l, k, j, i) = 1.0;
+          });
+
+      THEN("they should sum appropriately") {
+        using policy4D = Kokkos::MDRangePolicy<Kokkos::Rank<4>>;
+        Real total = 0.0;
+        Real sum = 1.0;
+        Kokkos::parallel_reduce(
+            policy4D({0, 0, 0, 0}, {v.GetDim(4), v.GetDim(3), v.GetDim(2), v.GetDim(1)}),
+            KOKKOS_LAMBDA(const int l, const int k, const int j, const int i,
+                          Real &vsum) { vsum += v(l, k, j, i); },
+            sum);
+        total += sum;
+        REQUIRE(std::abs(total - 20480.0) < 1.e-14);
+      }
+      AND_THEN("pulling out a subset by name should work") {
+        using policy4D = Kokkos::MDRangePolicy<Kokkos::Rank<4>>;
+        auto v = rc.PackVariables({"v2", "v3", "v5"});
+        Real total = 0.0;
+        Real sum = 1.0;
+        Kokkos::parallel_reduce(
+            policy4D({0, 0, 0, 0}, {v.GetDim(4), v.GetDim(3), v.GetDim(2), v.GetDim(1)}),
+            KOKKOS_LAMBDA(const int l, const int k, const int j, const int i,
+                          Real &vsum) { vsum += v(l, k, j, i); },
+            sum);
+        total += sum;
+        REQUIRE(std::abs(total - 16384.0) < 1.e-14);
+      }
+    }
+
+    WHEN("we set individual fields by index") {
+      PackIndexMap vmap;
+      auto v = rc.PackVariables(std::vector<std::string>({"v3", "v6"}), vmap);
+      const int iv3lo = vmap["v3"].first;
+      const int iv3hi = vmap["v3"].second;
+      const int iv6 = vmap["v6"].first;
+      par_for(
+          "Initialize variables", DevExecSpace(), 0, v.GetDim(3) - 1, 0, v.GetDim(2) - 1,
+          0, v.GetDim(1) - 1, KOKKOS_LAMBDA(const int k, const int j, const int i) {
+            v(1, k, j, i) = 1.0; // should correspond with iv3lo+1
```

Comment:
If variables are referenced by index/name within a kernel, it seems to me that a variable pack isn't needed or isn't the best way to write the loop. There are probably instances where it makes sense but I'm not seeing.

Reply:
@jdolence can comment more. But this is something we encountered when writing our own code. It's a convenient way to capture a bunch of variables in a lambda at once. I am rewriting this test a little to not implicitly assume index ordering.